v2/internal/reconcilers/arm/arm_client.go

```diff
@@ -33,3 +40,19 @@ func (c *armClient) GenericClient() *genericarmclient.GenericClient {
 func (c *armClient) CredentialFrom() string {
 	return c.credentialFrom.String()
 }
+
+// ARMClientDetails is the type returned by the ARMClientCache containing the actual client and some details about the destination
+// of the resource
+type ARMClientDetails struct {
```

Comment:
I am not in love with this type name.

It's not really _just_ `ClientDetails` it feels more like `DestinationDetails` or something? It has a client obviously, and also the target subscription, as well as the location that the credentials being used came from.

`AuthenticationDetails`?

It's a bit of a grab-bag but at the same time feels like an actual construct because all of the stuff is about:
1. Where we got the secret we're using.
2. What we read out of that secret (including a client constructed from the secret)

Reply:
Maybe `ARMConnectionDetails` or maybe just `ARMConnection` ?

Reply:
Yeah I like that more.

Since the module is ARM I actually dropped the prefix too and just called it `arm.Connection`

Reply:
Nice.